### ISSUE TITLE
Builder - Preview mode "Unable to create map: Cannot read property 'id' of null" + share issue has resurfaced

### ISSUE THREAD

**Issue by pjdohertygis** [Esri/geoform-template-js#130]
It looks like there are additional fixes needed to for issue #115 . If I (@kellyhutchins)  look at the code that was merged as part of this update the developer checks for the edit url parameter by looking for &edit and if its found strips it from the url. That works great but it looks like sometimes the url parameter is the first param so they’ll need to also check for ?edit. 

https://github.com/Esri/geoform-template-js/commit/4e49de765bfb0adc1f834a2dceb3e60cc56619d4

This is also causing issues for #119 because it tries to preview the app url with ?edit appended instead of the appid. 

Example of an app where the url param is ?edit=true
http://nw-brews.mapsdevext.arcgis.com/apps/GeoForm/index.html?edit=true&appid=42ad8156dc4e4d6e88b867aa0427d133&folderid=470b1c8357a048cf8e20066c3c42182e

Basically - now that we are in an ArcGIS Online hosted environment we need a fix ASAP.
